crates/engine/tree/src/tree/payload_processor/executor.rs

```diff
@@ -62,18 +69,23 @@ struct WorkloadExecutorInner {
 }
 
 impl WorkloadExecutorInner {
-    fn new(rayon_pool: rayon::ThreadPool) -> Self {
-        fn get_runtime_handle() -> Handle {
+    fn new(rayon_pool: rayon::ThreadPool, worker_threads: Option<usize>) -> Self {
+        fn get_runtime_handle(worker_threads: Option<usize>) -> Handle {
             Handle::try_current().unwrap_or_else(|_| {
-                // Create a new runtime if now runtime is available
+                // Create a new runtime if no runtime is available
                 static RT: OnceLock<Runtime> = OnceLock::new();
 
-                let rt = RT.get_or_init(|| Runtime::new().unwrap());
+                let rt = RT.get_or_init(|| match worker_threads {
+                    Some(num_of_threads) => {
+                        Builder::new_multi_thread().worker_threads(num_of_threads).build().unwrap()
+                    }
+                    None => Runtime::new().unwrap(),
+                });
```

Comment:
Hmm, I didn't consider that this style of initialization would not take into account any argument passed if this function were run a second time. I think for our purposes this is fine, since we should only ever initialize this once, in the engine. Let's document this behavior

Reply:
Done, added the documentation for the initialisation behaviour of `WorkloadExecutorInner`